pkg/client/token_interceptor.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"google.golang.org/grpc/metadata"
+
+	"google.golang.org/grpc"
+)
+
+// TokenInterceptor inject token from tokenservice if present and if provided context contain no one
+func (c *immuClient) TokenInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	if md, ok := metadata.FromOutgoingContext(ctx); !ok || len(md.Get("authorization")) == 0 {
+		if token, err := c.Tkns.GetToken(); err == nil && token != "" {
```

Comment:
The err is silently ignored here, I'm not sure if this is a good approach. Let's say that the file with token is not readable. In such situation the user will see authentication failure instead of IO error.